### ISSUE TITLE
[smartthings-tv] Error when getting devices: options must be an object

### ISSUE THREAD

**Issue by m-abdelhamid** [o-lukas/homebridge-smartthings-tv#331]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through common issues
- [X] Updated plugin to the latest version (if applicable)

### Describe the bug

Plugin was working beautifully until recent update. Now I get the error above when I restart homebridge and my device shows as 'No Response' in Home App and accessory has disappeared from homebridge. 

### To Reproduce

Any restart of homebridge. Tried enabling/disabling plugin; uninstall/re-install; new API Token. Also tried device mapping with Id and Mac address - no success.

### Expected behavior

Device (TV) would register and work.

### Logs

[smartthings-tv] Error when getting devices: options must be an object

### Logs

{
    "token": "XYXYXY-removed-XYXYXYX",
    "platform": "smartthings-tv"
}

### Environment Information

```markdown
- Plugin Version: v2.0.0
- Homebridge Version: v1.7.0
- Node.js version: v20.12.0
- NPM version: 10.5.0
- Operating System: macOS Catalina
```

**Comment by mrrust310** [comment]
I'm having the exact same issue. Happened after update.

**Comment by o-lukas** [comment]
Could you please upload your configuration and (if possible) the complete plugin logs. I can't really reproduce this error as it works fine for me. 

At first glance it looks like the Smartthings API package I'm using returns this error. But I have absolutely no idea why...

**Comment by mrrust310** [comment]
CONFIG:

{
    "platform": "smartthings-tv",
    "token": "b69e53b7-2538-4dce-a725-4b4173b3c9c9",
    "deviceMappings": [
        {
            "macAddress": "XX:XX:XX:XX:XX:XX",
            "deviceId": "c377ab8b-4955-5890-fb7c-4aa1edaXXXX"
        }
    ]
}

Log:

[log.txt](https://github.com/o-lukas/homebridge-smartthings-tv/files/14874582/log.txt)

Hope this helps

**Comment by o-lukas** [comment]
@mrrust310 Thanks for your information. I deleted your comment to prevent abuse of your API token.

**Comment by mrrust310** [comment]
Oh right, didn't notice that. Thanks

**Comment by o-lukas** [comment]
Your config looked ok to me. I tried using the same configuration as you have but with my API token, Mac address, etc. but everything did work as expected. I tried using Ubuntu and MacOS and different versions of Node but could not reproduce the error.

Could you please enable debug logging in the settings of your homebridge instance and provide a new log file. Maybe I can find a hint there...

**Comment by mrrust310** [comment]
I'm on Hoobs. The only debug login I see is this:

[log2.txt](https://github.com/o-lukas/homebridge-smartthings-tv/files/14876032/log2.txt)

**Comment by mrrust310** [comment]
I was able to get the plugin working again by restoring a backup with version 10.1.1.
This confirms the error is in the 2.0.0 update, which is what knocked it down.

Since it's working, I'm just going to leave like that. But please let me know if you figure this out so I can update to the latest.

**Comment by AlbertoAIG** [comment]
Hi,

I have exactly the same issue in macOS, but I had installed on a raspberry Pi and work perfetly with the versión 2.0.0. 
Mac and Raspberry have the same Node versión and I have tried reinstall and reconfigure the plugins and it doesn't work.

With previous version it doesn't work too.

This is my configuration:

```
{
    "token": "mytoken",
    "capabilityLogging": true,
    "registerPictureModes": true,
    "deviceBlacklist": [
        "mydevice1",
        "mydevice2"
    ],
    "_bridge": {
        "username": "0E:74:13:3B:7C:2C",
        "port": 35635
    },
    "platform": "smartthings-tv"
}

```
and the log:

```
[4/16/2024, 10:14:03 AM] [@o-lukas/homebridge-smartthings-tv] Launched child bridge with PID 75212
[4/16/2024, 10:14:04 AM] Registering platform '@o-lukas/homebridge-smartthings-tv.smartthings-tv'
[4/16/2024, 10:14:04 AM] [@o-lukas/homebridge-smartthings-tv] Loaded @o-lukas/homebridge-smartthings-tv v2.0.0 child bridge successfully
[4/16/2024, 10:14:04 AM] Loaded 0 cached accessories from cachedAccessories.0E74133B7C2C.
[4/16/2024, 10:14:04 AM] [@o-lukas/homebridge-smartthings-tv] Error when getting devices: options must be an object
[4/16/2024, 10:14:04 AM] Homebridge v1.7.0 (HAP v0.11.2) (@o-lukas/homebridge-smartthings-tv) is running on port 35635.

```
Thanks

**Comment by o-lukas** [comment]
Which version of Mac OS are you using? I will try to reproduce it again on the weekend.

**Comment by o-lukas** [comment]
> I was able to get the plugin working again by restoring a backup with version 10.1.1.
> 
> This confirms the error is in the 2.0.0 update, which is what knocked it down.
> 
> 
> 
> Since it's working, I'm just going to leave like that. But please let me know if you figure this out so I can update to the latest.

Compared to version 1.10.1 I only bumped some of the versions and removed support for Node Version 18 because of one dependency. But that should not be a problem since you are on Version 20 aren't you?

**Comment by AlbertoAIG** [comment]
> Which version of Mac OS are you using? I will try to reproduce it again on the weekend.

14.4.1 (last)

**Comment by mrrust310** [comment]
![image](https://github.com/o-lukas/homebridge-smartthings-tv/assets/115042099/92950b3d-36c4-4985-a1a6-5cbf2b469a4b)

**Comment by o-lukas** [comment]
Node version 16.x is not supported. Could you please try to update to Node version 18.x and check if this solves the problem?

**Comment by o-lukas** [comment]
> > Which version of Mac OS are you using? I will try to reproduce it again on the weekend.
> 
> 14.4.1 (last)

@AlbertoAIG I did test with this version of Mac OS and it did work as expected. Which version of node are you running?

**Comment by AlbertoAIG** [comment]
> > > Which version of Mac OS are you using? I will try to reproduce it again on the weekend.
> 
> > 
> 
> > 14.4.1 (last)
> 
> 
> 
> @AlbertoAIG I did test with this version of Mac OS and it did work as expected. Which version of node are you running?

20.12.2. With this version the plugin work on raspberry pi

**Comment by nishaon** [comment]
Hi, 
I am experiencing the same issue- **Error when getting devices: options must be an object**
It was working fine earlier.

I tried with different versions too without any luck.
Appreciate if you could assist as I really like this plugin.
Thanks.

Below for your info:
HomeBridge version in docker for Synology NAS: v1.8.0
Node.js Version | v18.18.2
Homebridge SmartThings TV plugin versions tried: 2.0.1, 2.0.0, 1.10.4., 1.10.1 & 1.10.0

Plugin Logs:
[4/21/2024, 11:01:01 AM] [@o-lukas/homebridge-smartthings-tv] Launched child bridge with PID 25034
[4/21/2024, 11:01:02 AM] Registering platform '@o-lukas/homebridge-smartthings-tv.smartthings-tv'
[4/21/2024, 11:01:02 AM] [@o-lukas/homebridge-smartthings-tv] Loaded @o-lukas/homebridge-smartthings-tv v2.0.1 child bridge successfully
[4/21/2024, 11:01:02 AM] Loaded 0 cached accessories from cachedAccessories.0E9486A287E2.
[4/21/2024, 11:01:02 AM] [@o-lukas/homebridge-smartthings-tv] Error when getting devices: options must be an object
[4/21/2024, 11:01:02 AM] Homebridge v1.8.0 (HAP v0.12.0) (@o-lukas/homebridge-smartthings-tv) is running on port 52175.
Failed to create listener for avahi-daemon server state. The system will not be notified about restarts of avahi-daemon and will therefore stay undiscoverable in those instances. Error message: Error: No such interface found


JSON Config (tried with the simplest one too):
{
    "token": "MyToken",
    "registerApplications": true,
    "deviceMappings": [
        {
            "ipAddress": "1"
        },
        {
            "deviceId": "MyDeviceID",
            "wol": true,
            "macAddress": "MyMAC",
            "ping": true,
            "ipAddress": "MyIP"
        }
    ],
    "_bridge": {
        "username": "0E:94:86:A2:87:E2",
        "port": 52175
    },

**Comment by anjos** [comment]
Same for me here.  Upgrading to v2.0.1 triggers the issue.  Staying on v2.0.0 works OK.

**Comment by anjos** [comment]
> Same for me here. Upgrading to v2.0.1 triggers the issue. Staying on v2.0.0 works OK.

Looking at what changed between these to minor releases, I'd say this is being triggered by bumped dependencies.

**Comment by huggi001** [comment]
Adding same here for me when updating to v2.0.1.  Going back to v2.0.0 works fine.

**Comment by speedingcheetah** [comment]
Same here. Update to 2.0.1 broke all functionality for me. Same error. Reverted to v2.0.0 works great.

**Comment by o-lukas** [comment]
I have been testing around with different versions of node and different operating systems but the error simply won't show up on my systems. I really don't know what's going on...

I have bumped some versions again as there have been updates. I hope this solves the problem but that would really just be a lucky shot to be honest 😅 Please check the [latest release v2.0.2](https://github.com/o-lukas/homebridge-smartthings-tv/releases/tag/v2.0.2)

Please let me know if it helps. If the error is still there, please reopen the issue and attach the following information about your environment:

- OS with exact version
- Node Version
- last Version of this plugin that works for you

I hope I can reproduce the error with that information and it will be no more guessing around. But let's just hope the problem solved itself with version 2.0.2...

**Comment by nishaon** [comment]
> I have been testing around with different versions of node and different operating systems but the error simply won't show up on my systems. I really don't know what's going on...
> 
> I have bumped some versions again as there have been updates. I hope this solves the problem but that would really just be a lucky shot to be honest 😅 Please check the [latest release v2.0.2](https://github.com/o-lukas/homebridge-smartthings-tv/releases/tag/v2.0.2)
> 
> Please let me know if it helps. If the error is still there, please reopen the issue and attach the following information about your environment:
> 
> * OS with exact version
> * Node Version
> * last Version of this plugin that works for you
> 
> I hope I can reproduce the error with that information and it will be no more guessing around. But let's just hope the problem solved itself with version 2.0.2...

Hi mate, thanks for releasing the update. It is not showing the error anymore with ver 2.0.2.
TV is showing in HomeBridge and can control from there.
However, I can't add the TV manually to HomeKit. I tried to add new accessory using the HomeKit code from child bridge, but it didn't detect the TV. 
Thanks.

**Comment by o-lukas** [comment]
> > I have been testing around with different versions of node and different operating systems but the error simply won't show up on my systems. I really don't know what's going on...
> > I have bumped some versions again as there have been updates. I hope this solves the problem but that would really just be a lucky shot to be honest 😅 Please check the [latest release v2.0.2](https://github.com/o-lukas/homebridge-smartthings-tv/releases/tag/v2.0.2)
> > Please let me know if it helps. If the error is still there, please reopen the issue and attach the following information about your environment:
> > 
> > * OS with exact version
> > * Node Version
> > * last Version of this plugin that works for you
> > 
> > I hope I can reproduce the error with that information and it will be no more guessing around. But let's just hope the problem solved itself with version 2.0.2...
> 
> Hi mate, thanks for releasing the update. It is not showing the error anymore with ver 2.0.2. TV is showing in HomeBridge and can control from there. However, I can't add the TV manually to HomeKit. I tried to add new accessory using the HomeKit code from child bridge, but it didn't detect the TV. Thanks.

Please try to reset externally published accessories in Homebridge using: Unpair Bridges / Cameras / TVs / External Accessories in the Homebridge settings. Pay attention that this can have effects on other plugins as well so use with caution.

**Comment by anjos** [comment]
I confirm the update is also working for me.
